### PR TITLE
Update to special member functions ddoc in c++ interop

### PR DESCRIPTION
**Pull request by Emmankoko**
This PR is an update to the special function header in C++ inter documentation. 
one direct change was to replace `special function` header to `operator overloading` since we have already talked about constructors and destructors.

this is a preview of how it is looking like now
please let me know any suggested changes, thanks!
https://dtest.dlang.io/artifact/website-5388b9dcb36ee1089914e5d68ff0da5d7a87f9a0-ec4cd8bc11f96e39d04ef037f2f1f3e5/web/spec/cpp_interface.html#operator-overloading

@dkorpel @Geod24

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: spec/cpp_interface.dd [hunk 1]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
     operators work in the two languages. For example, in D all objects are
     movable and there is no move constructor.)
 
-$(H2 $(LNAME2 special-member-functions, Special Member Functions))
-
-    $(P D cannot directly call C++ special member functions, and vice versa.
-    These include constructors, destructors, conversion operators,
-    operator overloading, and allocators.
+$(H2 $(LNAME2 operator-overloading, Operator Overloading))
+    $(P D's operator overloading is primarily involved with overloading its Unary
```
Comment (dkorpel) [inline spec/cpp_interface.dd:955]:
```suggestion
    $(P D's operator overloading is primarily involved with overloading its unary
```

File: spec/cpp_interface.dd [hunk 2]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+    and Binary operators including the increment, index, slice, cast, function call,
```
Comment (dkorpel) [inline spec/cpp_interface.dd:956]:
```suggestion
    and binary operators including the increment, index, slice, cast, function call,
```

File: spec/cpp_interface.dd [hunk 3]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+    comparison operator etc. overloads of these operators that exist in C++ can be
```
Comment (dkorpel) [inline spec/cpp_interface.dd:957]:
```suggestion
    comparison operator etc. Overloads of these operators that exist in C++ can be
```

File: spec/cpp_interface.dd [hunk 4]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+    int a;
+    int b;
+    int buffer[3];
+    A()
+    {
+        a = 0;
+        b = 0;
+    }
+    A(int a, int b);
+    A& operator=(const A& obj);//overloading the assignment operator
```
Comment (dkorpel) [inline spec/cpp_interface.dd:980]:
```suggestion
    A& operator=(const A& obj); //overloading the assignment operator
```

File: spec/cpp_interface.dd [hunk 5]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+    int& operator[](int value); // overloading the index operator
+    void printbuffer();
+};
+
+A::A(int a, int b)
+{
+    this->a = a;
+    this->b = b;
+}
+A& A::operator=(const A& obj)
```
Comment (dkorpel) [inline spec/cpp_interface.dd:991]:
```suggestion
}

A& A::operator=(const A& obj)
```

File: spec/cpp_interface.dd [hunk 6]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+}
+
+int& A::operator[](int value)
+{
+    return buffer[value];
+}
+
+void A::printbuffer()
+{
+    for(int i = 0; i < 3; i++)
```
Comment (dkorpel) [inline spec/cpp_interface.dd:1012]:
```suggestion
    for (int i = 0; i < 3; i++)
```

File: spec/cpp_interface.dd [hunk 7]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+    $(P In a bar.d file: )
+
+------
+import std.stdio;
+
+extern(C++) struct A
+{
+    int a;
+    int b;
+    int [3]buffer;
```
Comment (dkorpel) [inline spec/cpp_interface.dd:1028]:
```suggestion
    int[3] buffer;
```

File: spec/cpp_interface.dd [hunk 8]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+    this(int a, int b);
+    ref A opAssign(ref const A); //assignment operator interface
```
Comment (dkorpel) [inline spec/cpp_interface.dd:1030]:
```suggestion
    ref A opAssign(ref const A); // links with A::operator=
```

File: spec/cpp_interface.dd [hunk 9]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+    void opCall(int a, int b, int c); //function call operator interface
```
Comment (dkorpel) [inline spec/cpp_interface.dd:1031]:
```suggestion
    void opCall(int a, int b, int c); // links with A::operator()
```

File: spec/cpp_interface.dd [hunk 10]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+    ref int opIndex(int value); //index operator interface
```
Comment (dkorpel) [inline spec/cpp_interface.dd:1032]:
```suggestion
    ref int opIndex(int value); // links with A::operator[]
```

File: spec/cpp_interface.dd [hunk 11]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+    void printbuffer();
+}
+
+void main()
+{
+    auto obj1 = A(1, 3);
+    auto obj2 = A(13, 43);
+    obj2 = obj1; //opAssign called here
```
Comment (dkorpel) [inline spec/cpp_interface.dd:1040]:
```suggestion
    obj2 = obj1; // opAssign called here
```

File: spec/cpp_interface.dd [hunk 12]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+    writeln("obj2's a is ",obj2.a);
```
Comment (dkorpel) [inline spec/cpp_interface.dd:1041]:
```suggestion
    writeln("obj2's a is ", obj2.a);
```

File: spec/cpp_interface.dd [hunk 13]
```diff
@@ -951,13 +951,118 @@ $(H2 $(LNAME2 lifetime-management, Lifetime Management))
...
+    writeln("obj2's b is ", obj2.b);
+    obj2(20,40,60);
```
Comment (dkorpel) [inline spec/cpp_interface.dd:1043]:
```suggestion
    obj2(20, 40, 60);
```

**Review by dkorpel — APPROVED** [review]

**Comment by dkorpel** [comment]
Looks good! I just applied some style changes.

**Pull request merged**